openmls/src/group/mls_group/create_commit.rs

```diff
@@ -66,10 +72,41 @@ impl CoreGroup {
 
         let proposal_reference_list = proposal_queue.commit_list();
 
-        let sender_index = self.sender_index();
         // Make a copy of the current tree to apply proposals safely
         let mut diff: TreeSyncDiff = self.treesync().empty_diff()?;
 
+        // If this is not an external commit we have to set our own leaf index
```

Comment:
So this function is already 200 lines too long. Can you factor out the new things you put in here?